tools/python_api/src_cpp/py_connection.cpp

```diff
@@ -64,6 +65,18 @@ std::unique_ptr<PyQueryResult> PyConnection::execute(PyPreparedStatement* prepar
     return pyQueryResult;
 }
 
+std::unique_ptr<PyQueryResult> PyConnection::query(const std::string& statement) {
+    py::gil_scoped_release release;
+    auto queryResult = conn->query(statement);
+    py::gil_scoped_acquire acquire;
+    if (!queryResult->isSuccess()) {
```

Comment:
I refactored it for Python. In Node.js I think I'll just keep the two workers as different classes for simplicity instead of creating a common base class.